rebalance/rebalance.py

```diff
@@ -208,6 +210,16 @@ def rebalance(plugin, outgoing_scid, incoming_scid, msatoshi: Millisatoshi = Non
                         rpc_result = {'status':'error', 'message': 'No suitable routes found'}
                         return cleanup(plugin, label, payload, rpc_result)
                     continue
+
+            # build route cache and filter duplicates
+            route_str = ""
+            for hop in r['route']:
+                route_str += hop['channel'] + ' '
+            if routes.get(route_str) is not None:
+                rpc_result = {'status':'error', 'message': 'ROUTE DUP FOUND'}
+                return cleanup(plugin, label, payload, rpc_result)
+            routes[route_str] = True
```

Comment:
This should be unnecessary. If a route failed previously, then one of its components is in the `excludes` list. So there should be no duplicates anyway.
Or what I miss?

Reply:
Sorry, needed to push work-in-progress nonsense ... my machine broke down and I'm reinstalling now :D